### PR TITLE
Dire Troll Trait

### PR DESCRIPTION
**Pull request by ercarp**
## Changelog:
- Added a very basic **_Dire_** trait and a modifier to accompany it.
- Added localization and an icon for the trait.
- @zumbak04: Added code to ensure only Trolls generate with the Dire trait.
- Gave female trolls with the Dire trait a different portrait modifier (since female Dire Trolls look different).

## To-do:
- Consider restricting female trolls only to Nazmani culture?

> Female dire trolls can be **found within blood troll society**. Aka'ali the Conqueror was a female Zandalari dire troll princess.

Source: [Dire troll article on Wowpedia](https://wowpedia.fandom.com/wiki/Dire_troll)

- [x] Fix wc errors
- [x] Fix dire errors
- [x] Replace is_troll_trigger

## Screenshots:

<details><summary>Click to expand</summary>

image

</details>

# How to test:

1. Use `add_trait dire ID` [console command](https://ck3.paradoxwikis.com/Console_commands#Enabling_debug_mode) on any Troll character in-game.
2. Provide feedback on localization, trait icon, portrait modifier, and the attribute bonuses used on the trait.

### CODE REVIEW

File: common/traits/wc_genetic_traits.txt [hunk 1]
```diff
@@ -224,4 +226,41 @@ magic_good_noninheritable = {
 			desc = trait_magic_good_noninheritable_character_desc
 		}
 	}
+}
+
+dire = {
+	opposites = {
+		weak
```
Comment (zumbak04) [inline common/traits/wc_genetic_traits.txt:233]:
```suggestion
		weak
		physique_bad
		dwarf
		giant
```

File: common/traits/wc_genetic_traits.txt [hunk 2]
```diff
@@ -224,4 +226,41 @@ magic_good_noninheritable = {
...
+
+	prowess = 10
+	dread_baseline_add = 15
+	
+	health = -0.05
+
+	diplomacy = -1
+	stewardship = -1
+	intrigue = -1
+	learning = -1
```
Comment (zumbak04) [inline common/traits/wc_genetic_traits.txt:246]: I created a balancing card here. You can balance the modifiers as you want.
https://docs.google.com/spreadsheets/d/18zS-FAte6nsejx7rpzSmVuioRXps3iWSvSkx_xRJSWI/edit#gid=1768507286&range=A3:A6

**Review by TheGroxEmpire — COMMENTED** [review]
- It's possible to have Dire trait without having Trolls traits by interracial breeding. Since the flavor texts explicitly mentioned Trolls it's a bit iffy. Imo we should restrict it that only Trolls could get this trait.
<details>

image
</details>

**Comment by ercarp** [comment]
> It's possible to have Dire trait without having Trolls traits by interracial breeding. Since the flavor texts explicitly mentioned Trolls it's a bit iffy. Imo we should restrict it that only Trolls could get this trait.

This is already planned, I mentioned it in the main post. I don't know how to code it in myself but I've asked @zumbak04 to take a look at it.

**Review by TheGroxEmpire — APPROVED** [review]
> This is already planned, I mentioned it in the main post. I don't know how to code it in myself but I've asked @zumbak04 to take a look at it.

LGTM then!

**Review by TheDrPibb1999 — APPROVED** [review]
I think it all goods good from icon to their benefits, LGTM!

**Comment by ercarp** [comment]
What I was able to gather from the Wowpedia article regarding dire trolls:

- **especially large and powerful trolls**
  - This is already covered by not only making them physically larger but also increasing their Prowess stat.
- **dire trolls are not regarded as monstrous by other trolls**
  - No opinion penalty, basically. They still have some Natural Dread due to their imposing stature, but this only affects rulers anyway.
- **elite warriors of the Zandalari**
  - In the localization for the trait, I mentioned that they are "stand at the very top of the troll fighting elite", but this isn't specific to Zandalari. Maybe we could do something special for them? Not sure if necessary though.
- **short lived**
  - Going to either reduce health or life expectancy. A flat decrease to life expectancy might make more sense as I don't think dire trolls are weak, so they shouldn't die to something like wounds or illness.
- **many dire trolls have eyebrows, a trait not often seen in trolls**
  - Included in the trivia section in the article. But looking at photos of dire trolls, I don't actually see that many of them with eyebrows. If possible, I may add a chance for some dire trolls to generate with eyebrows to reflect this.
- **female dire trolls can be found within blood troll society**
  - This doesn't necessarily rule out the possibility of female dire trolls in other troll soecities, but if possible we could try restricting female dire trolls to Blood Trolls only.
- **[dire trolls provided] wisdom of war to those councilors who may seldom see a battlefront**
  - If dire trolls were able to provide "wisdom of war", I don't think they were stupid enough to warrant some of the stat decreases I had added. So I will amend these.

File: common/traits/wc_genetic_traits.txt [hunk 3]
```diff
@@ -224,4 +226,51 @@ magic_good_noninheritable = {
...
+	potential = {
+		OR = {
+			dire_possible_trigger = yes
+			mother = { dire_possible_trigger = yes }
+			real_father = { dire_possible_trigger = yes }
+		}
+	}
+	
+	birth = 0.5
+	random_creation = 0.5
```
Comment (zumbak04) [inline common/traits/wc_genetic_traits.txt:249]: It makes it so rare there are no dire trolls on Zandalar.
<details>
<summary>Click to expand</summary>

image

</details>

I would change it back to 5.

File: common/traits/wc_genetic_traits.txt [hunk 4]
```diff
@@ -224,4 +226,51 @@ magic_good_noninheritable = {
...
+
+	prowess = 13
+	dread_baseline_add = 15
+	martial = 8
+	diplomacy = -7
+	health = -1
+	enemy_hard_casualty_modifier = 0.15
```
Comment (zumbak04) [inline common/traits/wc_genetic_traits.txt:256]: This **GIANT** boost to martial makes this trait like the best martial education... plus `enemy_hard_casualty_modifier`. I don't think dire trolls are born geniuses of strategy. Moreover, the ones on Zandalar are pretty dummies. I would limit the bonus to Prowess and Dread. Prowess is enough to make them good commanders that don't die and great knights to make the army stronger.

File: common/traits/wc_genetic_traits.txt (hunk 3 above)
Comment (ercarp) [reply common/traits/wc_genetic_traits.txt]: > I would change it back to 5.

Changed in bd0dca9def81c5917ab248912b6b2b41613d2fae. 👻

File: common/traits/wc_genetic_traits.txt (hunk 4 above)
Comment (ercarp) [reply common/traits/wc_genetic_traits.txt]: Changed this in e3793d283bdb6dec711901887bee2473c807097c.

Removed everything except Prowess and Dread attributes, although I did increase Prowess from 13 to 15 to make up for the removed attributes. Updated the trait on the [balance sheet](https://docs.google.com/spreadsheets/d/18zS-FAte6nsejx7rpzSmVuioRXps3iWSvSkx_xRJSWI/edit#gid=1768507286).

File: common/traits/wc_genetic_traits.txt [hunk 5]
```diff
@@ -224,4 +225,49 @@ magic_good_noninheritable = {
...
+	
+	potential = {
+		OR = {
+			dire_possible_trigger = yes
+			mother = { dire_possible_trigger = yes }
+			real_father = { dire_possible_trigger = yes }
+		}
+	}
+	
+	birth = 0.5
```
Comment (zumbak04) [inline common/traits/wc_genetic_traits.txt:248]: The birth chance is the same as the random creation chance.
```suggestion
	birth = 5
```

File: common/traits/wc_genetic_traits.txt [hunk 6]
```diff
@@ -224,4 +225,49 @@ magic_good_noninheritable = {
 			desc = trait_magic_good_noninheritable_character_desc
 		}
 	}
-}
\ No newline at end of file
+}
+
+### Other
+dire = {
```
Comment (zumbak04) [inline common/traits/wc_genetic_traits.txt:231]: What about giving them the same opinion bonuses the Giant trait has?
<details>
<summary>Click to expand</summary>

image

</details>

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: common/traits/wc_genetic_traits.txt (hunk 6 above)
Comment (ercarp) [reply common/traits/wc_genetic_traits.txt:231]: Added in d88812f4c9f9999269af6c1029068c16fbfb586d.

**Review by zumbak04 — APPROVED** [review]
LGTM! Good job!

**Pull request merged**